### PR TITLE
enh: batch the fetching of authors

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

related to https://github.com/dust-tt/tasks/issues/387

We're making a looooot of these calls in Promise.all for workspaces that have many assistants

## Risk

inverted 80/20 type of PR (blast radius is whole site)

### CODE REVIEW

File: front/lib/api/assistant/recent_authors.ts [hunk 1]
```diff
@@ -123,46 +124,88 @@ function renderAuthors(
...
+  }, {} as Record<string, number[] | null>);
+
+  const authorByUserId: Record<number, UserTypeWithWorkspaces> = (
+    await getMembers(auth, {
+      userIds: Array.from(
+        new Set(Object.values(recentAuthorsIdsByAgentId).flat())
+      )
+        .filter((id) => id !== null)
+        // Ts doesn't narrow the type of `id` to `number` after the `filter` call.
+        .map((id) => id as number),
```
Comment (lasryaric) [inline front/lib/api/assistant/recent_authors.ts:178]:
```suggestion
        .flatMap((id) => id  ? [id] : []),
```

File: front/lib/api/assistant/recent_authors.ts [hunk 2]
```diff
@@ -123,46 +124,88 @@ function renderAuthors(
...
+    })
+  ).reduce((acc, member) => {
+    acc[member.id] = member;
+    return acc;
+  }, {} as Record<number, UserTypeWithWorkspaces>);
```
Comment (flvndvd) [inline front/lib/api/assistant/recent_authors.ts:183]: ✂️

File: front/lib/api/assistant/recent_authors.ts [hunk 3]
```diff
@@ -123,46 +124,92 @@ function renderAuthors(
...
   const currentUserId = auth.user()?.id;
 
-  const isGlobalAgent = versionAuthorId === null;
-  // For global agents, which have no authors, return early.
-  if (isGlobalAgent) {
-    return [DUST_OWNED_ASSISTANTS_AUTHOR_NAME];
-  }
+  const recentAuthorsIdsByAgentId: Record<string, number[] | null> = (
+    await Promise.all(
+      agents.map(async (agent): Promise<[string, number[] | null]> => {
```
Comment (flvndvd) [inline front/lib/api/assistant/recent_authors.ts:143]: Clearly not a big fan of map reduce 🙃.

**Review by flvndvd — APPROVED** [review]

File: front/lib/api/assistant/recent_authors.ts (hunk 3 above)
Comment (fontanierh) [reply front/lib/api/assistant/recent_authors.ts:143]: 🙃 🙃 🙃

**Pull request merged**